### DISCUSSION TITLE
Technical data collection in the Autoupgrade module

### DISCUSSION THREAD

**Discussion by gericfo** [PrestaShop/PrestaShop#36432] [Ideas]
### The problem

As you know, the aim of the Autoupgrade module (also known as the 1-Click upgrade module) is to ensure that users can upgrade their PrestaShop stores as easily as possible.

As maintainers of this module, we share the same goal. With this goal in mind, we'd like to make a number of improvements to the existing Autoupgrade module. 

Today, we're aware of some of the module's challenges and issues, thanks to the many feedbacks we've received from the community. However, we are still partly "blind" when it comes to the real usage of the module.

That's why we want to collect technical data on the Autoupgrade module. Capturing this data will guide our product decisions about the module.

### The solution

As you will understand, we want to collect technical data so that we can make the right decisions when it comes to optimizing the Autoupgrade module.

**How will technical data be collected?**
We will deploy the following data collection systems:
- **Product usage analytics -** Collect technical data regarding usage of the Autoupgrade module. This data will help us better understand user actions on the module (clicks on CTAs, upgrade channels used, number of successful upgrades, number of failed upgrades, etc.). Only technical data will be collected, no personal data or data enabling the identification of a physical person will be recovered.

- **Error reporting -** A technical report will be collected when an upgrade or rollback error occurs, based on a voluntary action triggered by the user (with a click on the "Report a bug" CTA). This data will help us better understand the nature of the errors (module version, PrestaShop version, PHP version, etc.).

**How will users be informed?**
We will update the "Data transparency" legal documentation of the PrestaShop project for the Autoupgrade module [link](https://www.prestashop-project.org/data-transparency/). A new section will be added, detailing the technical data collected, the use of this data, the retention period, the tools used and the opt-out options.
A link to this legal documentation will be included in the Autoupgrade module interface to alert all users.

**What opt-out systems will be available?**
Users will be offered several opt-out mechanisms to enable them to exercise their rights:
- Through a `PS_URL_TRACKING` variable to be set as `false`.
- By sending an email to the PrestaShop Privacy team.

**Who will have access to the technical data collected?**
Autoupgrade maintainers will have access to the technical data collected. Reports will be available on request to the Seamless Upgrade & Extensibility team.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
@gericfo 

Question 1 Will all maintainers get access to the data?
Question/Request 2 - I think it should be opt in, and the user should be able to set configure it in the GUI, not by changing a constant.

**Comment by gericfo** [comment]
Thanks for your feedback @Hlavtox!

Regarding your first point, all project maintainers will be able to access the data upon request. You will, for example, be able to access this data.

For your second point, direct opt-in is not required in this situation as we don't collect personal user data but only technical data about product usage. The project's data transparency documentation will be modified accordingly, and opt-out will be possible.
